src/test/java/uk/gov/justice/probation/courtcasematcher/controller/SqsAdminControllerIntTest.java

```diff
@@ -0,0 +1,85 @@
+package uk.gov.justice.probation.courtcasematcher.controller;
+
+import com.amazonaws.services.sqs.AmazonSQSAsync;
+import com.amazonaws.services.sqs.model.Message;
+import com.amazonaws.services.sqs.model.PurgeQueueRequest;
+import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
+import com.amazonaws.services.sqs.model.SendMessageRequest;
+import com.amazonaws.services.sqs.model.SendMessageResult;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.ActiveProfiles;
+import uk.gov.justice.probation.courtcasematcher.application.healthchecks.SqsCheck;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles({"local", "sqsadmin"})
+public class SqsAdminControllerIntTest {
+
+    @Autowired
+    private TestRestTemplate testRestTemplate;
+
+    @LocalServerPort
+    protected int port;
+
+    @MockBean
+    private SqsCheck sqsCheck;
+
+    @Autowired
+    @Qualifier("courtCaseMatcherSqsQueue")
+    private AmazonSQSAsync courtCaseMatcherSqsQueue;
+
+    @Autowired
+    @Qualifier("courtCaseMatcherSqsDlq")
+    private AmazonSQSAsync courtCaseMatcherSqsDlq;
+
+    @Value("${aws_sqs_court_case_matcher_endpoint_url}")
+    private String courtCaseMatcherSqsEndpointUrl;
+
+    @Value("${aws_sqs_court_case_matcher_dlq_endpoint_url}")
+    private String courtCaseMatcherSqsDlqEndpointUrl;
+
+    @Test
+    void givenThereAreMessagesOnDlq_whenRetryAllDlqInvoked_shouldReplayMessages() {
+
+        purgeQueues();
+
+        sendMessageToDlq("message body 1");
+        sendMessageToDlq("message body 2");
+
+        var response = testRestTemplate.exchange("http://localhost:" + port + "/retry-all-dlqs", HttpMethod.PUT, null, String.class);
```

Comment:
We generally use RestAssured rather than RestTemplate, there's an example [here](https://github.com/ministryofjustice/court-case-matcher/blob/9402aa735626ef579676e4c86c5e63d89df92b8b/src/test/java/uk/gov/justice/probation/courtcasematcher/health/HealthCheckTest.java#L51). Tends to read a bit better IMO but we should probably stick with it for consistency.